plugins/transformer/basic/client/recordupdater/record_updater.go

```diff
@@ -158,6 +158,54 @@ func (r *RecordUpdater) ObfuscateSensitiveColumns() (arrow.Record, error) {
 	return r.ObfuscateColumns(sensitiveColumnsArr)
 }
 
+func (r *RecordUpdater) DropRows(columnNames []string, value string) (arrow.Record, error) {
+	cols := r.record.Columns()
+
+	rowsToDrop := make([]int, 0, r.record.NumRows())
+	for j, column := range cols {
+		if !slices.Contains(columnNames, r.record.ColumnName(j)) {
+			continue
+		}
+		for i := range column.Len() {
+			if slices.Contains(rowsToDrop, i) {
+				continue
+			}
+			if (!column.IsValid(i) && value == "") || (column.IsValid(i) && column.ValueStr(i) == value) {
```

Comment:
`column.IsValid` means a `nil` value, don't we want to distinguish `nil` from empty string?

Reply:
If a user wanted to drop rows with a `nil` value how would they represent that in the yaml?

Reply:
Converted the `value` to take a `*string` so we can tell the difference between empty string and nil